Sources/Features/Client/BacktraceOomWatcher.swift

```diff
@@ -0,0 +1,148 @@
+import Foundation
+
+final class BacktraceOomWatcher {
+    private var applicationState = [String :Any]()
+    
+    
+    private(set) var lowMemoryFilePrefix = "_lowMemory"
+    private(set) static var oomFilePath: URL? = getStatusFilePath()
+    
+    private(set) var statusFilePath: URL
+    private(set) var crashReporter: CrashReporting
+    private(set) var attributesProvider:AttributesProvider
+    private(set) var backtraceApi: BacktraceApi
+    private let repository: PersistentRepository<BacktraceReport>
+    
+    init(repository:  PersistentRepository<BacktraceReport>, crashReporter: CrashReporting, attributes: AttributesProvider, backtraceApi: BacktraceApi) {
+        self.crashReporter = crashReporter
+        self.attributesProvider = attributes
+        self.backtraceApi = backtraceApi
+        self.repository = repository
+        
+        // set default state
+        self.applicationState["state"] = "foreground"
+        self.applicationState["debugger"] = DebuggerChecker.isAttached()
+        self.applicationState["appVersion"] = Bundle.main.object(forInfoDictionaryKey: kCFBundleVersionKey as String)
```

Comment:
Is this casting to `String` necessary?

Reply:
No, it isn't. Good catch! Thanks :)

Reply:
oh actually it is - kCFBundleVersionKey is CString. - we need to cast it to String I guess?